src/command/list/plain.rs

```diff
@@ -346,17 +354,13 @@ mod tests {
         fn single_project() {
             assert_eq!(
                 describe_packages(&[Package::Project {
-                    details: PackageDetails {
-                        name: "typescript".into(),
-                        version: TYPESCRIPT_VERSION.clone(),
-                    },
+                    name: "typescript".into(),
                     path: PROJECT_PATH.clone(),
-                    node: NODE_VERSION.clone(),
                     tools: vec!["tsc".into(), "tsserver".into()]
                 }])
                 .expect("Should always return a `String` if given a non-empty set")
                 .as_str(),
-                "package typescript@3.4.1 / tsc, tsserver / node@12.4.0 npm@built-in (current @ /a/b/c)"
+                "package typescript / tsc, tsserver / (current @ /a/b/c)"
```

Comment:
A couple notes here:

**First,** one of the design constraints for the `--format=plain` output was that it be *consistent* for each "type"—so all `package` listings have the same layout, and so on. This changes that subtly, in that you can now end up with `package typescript / ...` *or* `package typescript@<version> / ...`. I think this part is fine, though, because the idea was that the combination of the "type" at the beginning (`package`) and then splitting on the section separators (` / `) would be unambigous, and this hasn't changed that. 👍

**Second,** I suggest we supply `@project` as the version in this case instead of dropping it entirely. Doing so would mean that anyone parsing the string would still have the same *shape* to parse; they would just need to add a new case to handle: SemVer *or* `project`. When we fix it, they don't have to change *anything* as long as they didn't *delete* their SemVer parsing. (We can communicate that they *shouldn't* in our release notes/blog post/etc.)

The output, in this proposal, would end up looking like:

```
package typescript@project / tsc, tsserver / node@project npm@project (current @ /a/b/c)
```

This isn't the prettiest thing in the universe, but it remains unambiguous, is a minimal change for existing code and straightforward to add support for, and minimizes the change that users have to do later.

Reply:
That makes sense, I'd also be interested to hear from @dherman and @mikrostew their thoughts on the design.